### ISSUE TITLE
Replace core/VFS.py with PyFilesystem

### ISSUE THREAD

**Issue by alvanson** [fofix/fofix#108]
In an effort to simplify the code base, I recommend switching out or reimplementing `fofix.core.VFS` with [PyFilesystem](https://www.willmcgugan.com/blog/tech/post/creating-a-virtual-filesystem-with-python-and-why-you-need-one/). PyFilesystem achieves the same goals as `VFS` but is actively maintained. This also has the advantage of reducing the complexity of the FoFiX code base (but, of course, adds dependencies).

**Comment by nhydock** [comment]
Following the chain of links, should the switch be to PyFilesystem2, as is even recommended on their official repository?
https://github.com/PyFilesystem/pyfilesystem2

**Comment by alvanson** [comment]
Yes, sorry, meant PyFilesystem2. Their website is inconsistent with the '2' branding.

**Comment by Linkid** [comment]
I agree with you, I would like to remove this VFS script too because it is to complex for what we need to do.

**Comment by mdsitton** [comment]
Well replacing 1 file with a project that is much larger we will likely end up with more features than what we have now, so complexity vs what we do is not a great measure of things.

Regardless, i'd be up for switching. +1
